biocypher/write/graph/_neo4j.py

```diff
@@ -22,13 +24,45 @@ class _Neo4jBatchWriter(_BatchWriter):
         - _write_array_string
     """
 
+    def __init__(self, *args, **kwargs):
+        """
+        Constructor.
+
+        Check the version of Neo4j and adds a command scope if version >= 5.
+
+        Returns:
+            _Neo4jBatchWriter: An instance of the writer.
+        """
+
+        # Should read the configuration and setup import_call_bin_prefix.
+        super().__init__(*args, **kwargs)
+
+        try:
+            neo4j_version = _n4jversion.Neo4jVersion()
+            neo4j_version = neo4j_version.version
+        except:
+            logger.info(
+                "Not able to determine Neo4j version. Use default (version 4)."
+            )
+            neo4j_version = 4
```

Comment:
Great work so far @mbaric758 !
After carefully reviewing both the online and offline cases, I noticed one aspect that I think we should address together:
- When it comes to the online mode, everything seems to be running smoothly. The neo4j database version is directly fetched from the driver, and the correct import call is utilized. So, no issues there.
- However, in the offline mode, the version determination becomes a bit tricky. At the time of creating the Knowledge Graph (KG), we might not know whether the user intends to insert the KG into Neo4j v4 or v5. To make things more user-friendly, it would be advantageous if the generated import call could determine the version at runtime and use the appropriate command.

To tackle this, I could imagine the following:
- We could integrate an if-else condition within the `neo4j-admin-import.sh` script, allowing it to dynamically choose the appropriate command based on the detected version.
- Alternatively, we could create two separate scripts: `neo4j-admin-import.sh` and `neo4j-v5-admin-import.sh`, each tailored to handle the import process for the respective versions.

What do you think about this @mbaric758 ?

Reply:
Hi @nilskre , thanks for the review! 

Probably the most user friendly way of doing it would be to go with your first suggestion - implementing an if-else condition within `neo4j-admin-import.sh` , so the whole process of import call generation is automatic based on Neo4j version.

Reply:
The issue with neo4j-utils is now fixed (by using the latest working version `0.0.7` and implementing the rest directly in BioCypher).
Furthermore, it is now implemented that both import calls for v4 and v5 are written into the `neo4j-admin-import.sh`. At the moment there is not yet a if else in the import script.

So we are on the home straight.
The last open points are:
- Implement the if else in the script?
- Adapt the tests in `test/write/graph/test_neo4j.py` to test if the import script is correct for v4 and v5.
- Update the [neo4j docs](https://biocypher.org/output/neo4j.html) according to the changes.

@mbaric758 Can you verify my changes and have a look at open points?

Thanks a lot in advance!

Reply:
Thanks @nilskre ! I implemented an if-else statement for the import script which checks Neo4j version and then uses the appropriate import call. It doesn't seem to change the way the tests behave in `test/write/graph/test_neo4j.py` . Have a look and tell me what you think. Thanks!

Reply:
Regarding the tests: E.g. [here](https://github.com/biocypher/biocypher/blob/main/test/write/graph/test_neo4j.py#L56) and [here](https://github.com/biocypher/biocypher/blob/main/test/write/graph/test_neo4j.py#L110) we assert, how the import script should look like. In [this commit](https://github.com/biocypher/biocypher/pull/321/commits/c85e52f11c667195476b68753032cbe1e0f256b8) you have adapted this part of the tests to pass for the new adapted import call. It would be great, if you can adapt this to check if the import call looks like expected for v4 and v5 (all in one script). At the moment it only checks if some general words are in the import call.

Reply:
I added a test to `test/write/graph/test_neo4j.py` that checks whether all the distinct elements of both import calls are present within the generated import call, so have a look and tell me what you think.

Reply:
Looks good, thanks!